subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/pkg/webserver/webserver.go

```diff
@@ -254,3 +257,44 @@ func (ws *WebServer) serveAffinityGroups(w http.ResponseWriter, r *http.Request)
 		"NotImplemented: %v: %v",
 		r.Method, r.URL.Path)))
 }
+
+func (ws *WebServer) serveClusterStatus(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodGet {
+		w.Write(common.ToJsonBytes(ws.iHandlers.GetClusterStatusHandler()))
+		return
+	}
+
+	panic(internal.NewBadRequestError(fmt.Sprintf(
+		"NotImplemented: %v: %v",
+		r.Method, r.URL.Path)))
+}
+
+func (ws *WebServer) servePhysicalClusterStatus(w http.ResponseWriter, r *http.Request) {
+	if r.Method == http.MethodGet {
+		w.Write(common.ToJsonBytes(ws.iHandlers.GetPCStatusHandler()))
+		return
+	}
+
+	panic(internal.NewBadRequestError(fmt.Sprintf(
+		"NotImplemented: %v: %v",
+		r.Method, r.URL.Path)))
+}
+
+func (ws *WebServer) serveVirtualClustersStatus(w http.ResponseWriter, r *http.Request) {
+	name := strings.TrimPrefix(r.URL.Path, si.VirtualClustersPath)
+	if name == "" {
+		if r.Method == http.MethodGet {
+			w.Write(common.ToJsonBytes(ws.iHandlers.GetAllVCsStatusHandler()))
```

Comment:
>GetAllVCsStatusHandler [](start = 43, length = 22)

Also rename it?
serveVCsStatus
GetVCStatus
GetAllVCsStatusHandler
etc

to be VirtualClusterXXX?